software-templates/migrate-repo-to-enterprise/template.yaml

```diff
@@ -0,0 +1,300 @@
+apiVersion: scaffolder.backstage.io/v1beta3
+kind: Template
+metadata:
+  name: GitHub-Repo-Transfer
+  title: Transfer a repository between organizations
+  description: Transfer a repository from the bcgov GitHub organization to the bcgov-public GitHub enterprise organization
+spec:
+  owner: "developer-experience"
+  type: service
+
+  parameters:
+    - title: Transfer a repository
+      required:
+        - repoToTransfer
+        - owner
+        - destination
+      properties:
+        repoToTransfer:
+          title: Repository to transfer
+          type: string
+          ui:field: RepoUrlPicker
+          ui:options:
+            requestUserCredentials: # this is super important - it causes the user to be prompted to log in to GitHub and/or authorize the required GH scopes.
+              secretsKey: USER_OAUTH_TOKEN
+              additionalScopes:
+                github:
+                  - workflow
+            allowedHosts:
+              - github.com
+            allowedOwners:
+              - bcgov
+        destination:
+          title: Destination GitHub Organization
+          type: string
+          description: GitHub organization where the repo will be transferred to.
+          enum: 
+            - bcgov-public
+          enumNames: 
+            - "bcgov-public"
+        owner:
+          title: GitHub team (required)
+          type: string
+          description: The team in the bcgov-public GitHub Organization that maintains the repo.
+          ui:field: OwnerPicker
+          ui:options:
+              allowArbitraryValues: false
+              catalogFilter:
+                - kind: [ Group ]
+                  metadata.namespace: bcgov-public
+
+    - title: Provide information about your product/service/component
+      required:
+        - title
+        - description
+        - type
+        - lifecycle
+      properties:
+        title:
+          title: Title
+          type: string
+          description: A title for your product. For example, "Whizzbang Service".
+          ui:autofocus: true
+        description:
+          title: Description
+          type: string
+          description: A description for product/service/component.
+        # type field for catalog-info.yaml file, these values based on https://backstage.io/docs/features/software-catalog/descriptor-format/#spectype-required
+        type:
+          title: Product type
+          type: string
+          description: What your product/service/component provides.
```

Comment:
Need some refinement text here. Maybe "Which categorization best matches the contents of the repository?"

Reply:
Updated text with the suggested phrasing. Also renamed the title "Product type" to "Product category"